spraycharles.py

```diff
@@ -73,20 +73,36 @@ def args(passlist, userlist, host, module, path, csvfile, attempts, interval, eq
         colors.color_print("Must set --path to use the NTLM authentication module", colors.red)
         exit()
 
-    return users, passwords, passlist, userlist, host, module, path, csvfile, attempts, interval, equal, timeout, port, fireprox, domain, analyze_results, jitter, jitter_min 
+    if notify and webhook is None:
+        colors.color_print("Must specify a Webhook URL when the notify flag is used.", colors.red)
+        exit()
+
+    return users, passwords, passlist, userlist, host, module, path, csvfile, attempts, interval, equal, timeout, port, fireprox, domain, analyze_results, jitter, jitter_min, notify, webhook, pause
 
-def check_sleep(login_attempts, attempts, interval, csvfile, analyze_results):
+def check_sleep(login_attempts, attempts, interval, csvfile, analyze_results, notify, webhook, host, pause):
     if login_attempts == attempts:
         if analyze_results:
-            analyzer = analyze.Analyzer(csvfile)
-            analyzer.analyze()
+            analyzer = analyze.Analyzer(csvfile, notify, webhook, host)
+            success = analyzer.analyze()
+
+        # Pausing if specified by user before continuing with spray
+        if success is True and pause:
```

Comment:
`success` var won't be defined and will error unless `--analyze` thrown. Maybe indent this `if` block so it falls under the above`analyze_results` check